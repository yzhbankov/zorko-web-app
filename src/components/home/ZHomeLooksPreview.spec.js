import {mount, shallow} from 'vue-test-utils'
import {createRenderer} from 'vue-server-renderer'
import ZHomeLooksPreview from './ZHomeLooksPreview.vue'

describe('ZHomeLooksPreview.vue', () => {
  let defaultProps

  beforeEach(() => {
    defaultProps = {
      src: 'url/to/image.png',
      title: 'Look Title',
      path: 'path to look',
      author: {
        login: 'foo',
        avatar: 'foo/baz.png'
      },
      datum: {
        name: 'ddd'
      }
    }
  })

  it('initialize', () => {
    const wrapper = mount(ZHomeLooksPreview, {
      propsData: defaultProps
    })

    expect(wrapper.vm).toBeTruthy()
  })

  it('emits `openLook` when click on title', () => {
    const wrapper = mount(ZHomeLooksPreview, {
      propsData: defaultProps
    })

    wrapper.find('.js-ZHomeLooksPreview-title-parent').trigger('click')

    expect(wrapper.emitted('openLook')[0][0]).toEqual({
      title: defaultProps.title,
      path: defaultProps.path
    })
  })

  it('emits `openLook` when click on image', () => {
    const wrapper = mount(ZHomeLooksPreview, {
      propsData: defaultProps
    })

    wrapper.find('.js-ZHomeLooksPreview-image').trigger('click')

    expect(wrapper.emitted('openLook')[0][0]).toEqual({
      title: defaultProps.title,
      path: defaultProps.path

    })
  })

  it('emits `openDatum` when click in datum', () => {
    const wrapper = mount(ZHomeLooksPreview, {
      propsData: defaultProps
    })

    wrapper.find('.js-ZHomeLooksPreview-datum').trigger('click')

    expect(wrapper.emitted('openDatum')[0][0]).toEqual({
      name: defaultProps.datum.name
    })
  })

  it('has same HTML structure', () => {
    const renderer = createRenderer()
    const wrapper = shallow(ZHomeLooksPreview, {
      propsData: defaultProps
    })

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
