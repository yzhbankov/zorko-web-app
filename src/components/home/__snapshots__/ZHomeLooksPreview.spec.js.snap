// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZHomeLooksPreview.vue has same HTML structure 1`] = `
<v-card tile="">
    <v-card-media src="url/to/image.png" height="270px" width="361px" class="util-pointer js-ZHomeLooksPreview-image"></v-card-media>
    <v-card-title class="util-pointer js-ZHomeLooksPreview-title-parent">
        <div>
            <h6 class="grey--text util-bold">Look Title</h6>
        </div>
    </v-card-title>
    <v-card-actions class="white">
        <v-chip class="util-pointer">
            <v-avatar>
                <img src="foo/baz.png" alt="trevor">
            </v-avatar>
            foo
        </v-chip>
        <v-spacer></v-spacer>
        <v-chip class="util-pointer js-ZHomeLooksPreview-datum">
            <v-avatar>
                <v-icon class="green white--text">call_merge</v-icon>
            </v-avatar>
            ddd
        </v-chip>
    </v-card-actions>
</v-card>
`;
